go.mod

```diff
@@ -1,36 +1,38 @@
 module github.com/digitalocean/do-operator
 
-go 1.21
+go 1.22.0
```

Comment:
Should this match the toolchain value ?

Reply:
nope, not necessarily; when we specify `go 1.22` and run `go mod tidy`, this is what we get by default. this is how they are for many other repos with `go 1.22` update (ex. [CCM go.mod](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/master/go.mod))

here are the two docs I have on go toolchain: https://go.dev/doc/toolchain (not as helpful as the other doc) and https://news.ycombinator.com/item?id=36455694 (the first answer). AFAIU the `go` directive is still like the minimum go toolchain version while the `toolchain` directive is used to specify the precise compiler version that should be used..?